core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -310,9 +307,9 @@ class KafkaApis(val requestChannel: RequestChannel,
           if (topicPartition.topic == GROUP_METADATA_TOPIC_NAME
               && partitionState.deletePartition) {
             val leaderEpoch = if (partitionState.leaderEpoch >= 0)
-              Some(partitionState.leaderEpoch)
```

Comment:
What was the reasoning for this change? Are we trying to move over to java?

Reply:
Ah I see the interface is java. Are we planning to move everything to Java though? It's a bit tricky since the GroupCoordinatorAdaptor was done in scala.

Reply:
The interface is in Java and the new group coordinator will be in Java as well.

Reply:
Are we going to get rid of the adapter in scala too at some point then?

Reply:
The entire group coordinator in scala will disappear at some point, including the adapter.